### PR TITLE
bpo-39350: Fix fractions for int subclasses

### PR DESCRIPTION
**Pull request by vstinner**
Fix regression in fractions.Fraction if the numerator and/or the
denominator is an int subclass.

https://bugs.python.org/issue39350

### CODE REVIEW

**Comment by vstinner** [comment]
That's an alternative fix for https://bugs.python.org/issue39350

See PR #18309 to discussion (the other proposed fix).

**Comment by vstinner** [comment]
I tested manually that fractions.Fraction works again with numpy.int64: it works as expected.

See: https://bugs.python.org/issue39350#msg361483

File: Misc/NEWS.d/next/Library/2020-02-06-13-34-52.bpo-39350.wRwup1.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Fix regression in :class:`fractions.Fraction` if the numerator and/or the
```
Comment (mdickinson) [inline Misc/NEWS.d/next/Library/2020-02-06-13-34-52.bpo-39350.wRwup1.rst:1]: I think there's more to say here.

This _does_ fix a regression from 3.9.0a2 to 3.9.0a3, but it also represents a behaviour change from 3.8.x to 3.9.x. We should document that behaviour change, at least in the news entry.

**Review by mdickinson — APPROVED** [review]
The change LGTM, but we should make it clear in the Misc/NEWS entry that there's an actual behaviour change here from 3.8: it's not simply a case of a regression being introduced and then fixed.

**Comment by vstinner** [comment]
@mdickinson: I completed the documentation, would you mind to review it?

@mdickinson: You approved my PR. Does it mean that you are ok to always divide the numerator and denominator with an int, which is a Python 3.9 incompatible change? I'm fine with that, since I propose the change. But I prefer to confirm with you, since you proposed PR #18309 :-)

File: Doc/library/fractions.rst [hunk 2]
```diff
@@ -84,6 +84,10 @@ another rational number, or from a string.
       The :class:`Fraction` constructor now accepts :class:`float` and
       :class:`decimal.Decimal` instances.
 
+   .. versionchanged:: 3.9
+      The :func:`math.gcd` function is now used to normalize the *numerator*
+      and *denominator*. :func:`math.gcd` always return a :class:`int` type.
+      Previously, the GCD type depended in *numerator* and *denominator*.
```
Comment (mdickinson) [inline Doc/library/fractions.rst]: Typo: "on" rather than "in"

**Comment by mdickinson** [comment]
Still LGTM, especially with the `versionchanged` update. Thank you!

>  Does it mean that you are ok to always divide the numerator and denominator with an int, which is a Python 3.9 incompatible change? 

@vstinner Yes, I'm personally fine with that change; I don't know how other interested parties might feel about it, and there seems a risk that people won't even notice that there's a behaviour change proposed and implemented without a dedicated b.p.o. issue. OTOH, I think it _ought_ to be relatively uncontroversial, so perhaps we don't need that issue and discussion.

File: Doc/library/fractions.rst (hunk 2 above)
Comment (vstinner) [reply Doc/library/fractions.rst]: Oops, right, it's a typo. It's now fixed. Thanks for the review.

**Comment by vstinner** [comment]
> @vstinner Yes, I'm personally fine with that change; I don't know how other interested parties might feel about it, and there seems a risk that people won't even notice that there's a behaviour change proposed and implemented without a dedicated b.p.o. issue. OTOH, I think it ought to be relatively uncontroversial, so perhaps we don't need that issue and discussion.

I really liked the discussion :-) It was required to discuss how other types are handled. At the beginning, I didn't think about types other than int. At least, my PR now adds a test for an int subclass :-)

**Pull request merged**